exist-core/src/main/java/org/exist/xquery/functions/map/MapExpr.java

```diff
@@ -77,14 +77,23 @@ public void accept(final ExpressionVisitor visitor) {
     @Override
     public void dump(final ExpressionDumper dumper) {
         dumper.display("map {");
-        for (final Mapping mapping : this.mappings) {
+        for (int i = 0; i < this.mappings.size(); i++) {
+            final Mapping mapping = this.mappings.get(i);
+            if (i > 0) {
+                dumper.display(", ");
+            }
             mapping.key.dump(dumper);
-            dumper.display(" := ");
+            dumper.display(" : ");
             mapping.value.dump(dumper);
         }
         dumper.display("}");
```

Comment:
I could imagine using a single boolean would remove the total amount of additional calls eventually:
```
boolean firstMapping = true;
for (final Mapping mapping : this.mappings) {
    mapping.key.dump(dumper);
    dumper.display(" := ");
    dumper.display(" : ");
    mapping.value.dump(dumper);
    if (firstMapping) {
      firstMapping = false;
    } else {
        dumper.display(", ");
    }
}
```

Reply:
If I am not mistaken you code will end up with a trailing `,` as it postfixes with the comma character, whereas my code prefixes with the comma character to ensure no "dangling" comma.

Reply:
Sorry, the if/else block should of course go prior the key/value rendering...